test/Scientist.Test/FireAndForgetResultPublisherTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub;
+using NSubstitute;
+using UnitTests;
+using Xunit;
+
+[Collection("Tests Dependent Upon static ResultPublisher that can't run in parallel")]
+public class FireAndForgetResultPublisherTests
+{
+    [Fact]
+    public async Task PublishesAsynchronously()
+    {
+        const int expectedResult = 42;
+
+        // Create a new publisher that will delay all
+        // publishing to account for this test.
+        var innerPublisher = Substitute.For<IResultPublisher>();
+        innerPublisher.Publish(Arg.Any<Result<int, int>>())
+            .Returns(call => Task.Delay(100));
+
+        var fireAndForgetPublisher = new FireAndForgetResultPublisher(innerPublisher, ex => { });
+
+        var mock = Substitute.For<IControlCandidate<int, string>>();
+        mock.Control().Returns(expectedResult);
+        mock.Candidate().Returns(expectedResult);
+
+        const int count = 10;
+        using (Swap.Publisher(fireAndForgetPublisher))
+        {
+            Parallel.ForEach(
+                Enumerable.Repeat(0, count),
+                src =>
+                {
+                    var result = Scientist.Science<int>("myExperiment", experiment =>
+                    {
+                        experiment.Use(mock.Control);
+                        experiment.Try(mock.Candidate);
+                    });
+
+                    Assert.Equal(expectedResult, result);
+                });
+        }
+
+        // Make sure that the above science calls are still publishing.
+        Task whenPublished = fireAndForgetPublisher.WhenPublished();
+        Assert.NotNull(whenPublished);
+
+        // Ensure that the mock was called before the when published task has completed.
+        mock.Received(count).Control();
+        mock.Received(count).Candidate();
+
+        Assert.False(whenPublished.IsCompleted, "When Published Task completed early.");
+
+        await whenPublished;
+
+        Assert.True(fireAndForgetPublisher.WhenPublished().IsCompleted, "When Published Task isn't complete.");
```

Comment:
Shouldn't we just assert `whenPublished.IsCompleted` here? As in:

`Assert.True(whenPublished.IsCompleted, "When Published Task isn't complete.");`